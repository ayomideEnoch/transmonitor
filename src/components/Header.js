import React from 'react';
import PropTypes from 'prop-types';
import {
  Badge,
  Navbar,
  NavbarBrand,
} from 'reactstrap';

export function Header(props) {
  return props.latestBlock ? (
    <Navbar color="dark" dark expand="lg">
      <NavbarBrand href="#transactions" onClick={() => props.selectBlock(props.latestBlock)}>Transactions Monitor <Badge color="info" tooltip="Latest Block">#{props.latestBlock}</Badge>
    </NavbarBrand>
  </Navbar>
  ): (
  <Navbar color="dark" dark expand="lg">
    <NavbarBrand href="#">Transactions Monitor</NavbarBrand>
  </Navbar>
  );
}

Header.propTypes = {
  latestBlock: PropTypes.number,
  selectBlock: PropTypes.func.isRequired,
};

export default Header;
