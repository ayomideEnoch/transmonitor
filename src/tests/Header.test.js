import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Header from '../components/Header';

describe('BlockList', function() {
  let mockProps = {
    latestBlock: 123456,
    selectBlock: () => {},
  };

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Header {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('render correctly', () => {
    const component = renderer.create(<Header {...mockProps} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
