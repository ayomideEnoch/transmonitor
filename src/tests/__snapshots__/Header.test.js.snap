// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockList render correctly 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    href="#transactions"
    onClick={[Function]}
  >
    Transactions Monitor 
    <span
      className="badge badge-info"
      tooltip="Latest Block"
    >
      #
      123456
    </span>
  </a>
</nav>
`;
